addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -2088,3 +2088,40 @@ def test_passes_partner_id_to_moves_without_package_ids_at_creation_time(self):
 
         self.assertEqual(len(new_test_picking_pick.move_lines), 2)
         self.assertEqual(new_test_picking_pick.mapped("move_lines.partner_id"), partner)
+
+class TestStockPickingFloatQty(TestStockPickingCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestStockPickingCommon, cls).setUpClass()
+        # Product lot tracked in meters
+        uom_meter = cls.env.ref("uom.product_uom_meter")
+        cls.cable = cls.create_product("Cable", tracking="lot", uom_id=uom_meter.id, uom_po_id=uom_meter.id)
+
+    def test_multiple_receive_in_fractional_qty(self):
+        """ Receive a lot tracked product in fractional quantities to ensure no rounding error
+        occurs.
+        Scenario: Goods in for 1.2 m. Product received in 12 batches of 0.1 m. Rounding error could
+        change 0.1 to 0.11 and split to extra move line with qty 0.01 and block validating the pick
+        """
+        product_info = [{"product": self.cable, "uom_qty": 1.2}]
+        picking = self.create_picking(
+            self.picking_type_goods_in, products_info=product_info, assign=True
+        )
+
+        product_ids = [{"barcode": "productCable", "uom_qty": 1.2,
+                        "lot_names": [f"l:02" for l in range(12)],      #["01", "02",..., "12"]
```

Comment:
We could just do this? Similarly for `lot_quantities` and in the asserts.
```python
['1.02] * 12
```

Reply:
Currently this would be:
```
>>> f"l:02" for l in range(12)]
['l:02', 'l:02', 'l:02', 'l:02', 'l:02', 'l:02', 'l:02', 'l:02', 'l:02', 'l:02', 'l:02', 'l:02']
```
I think it's just missing {}, and needs slicing up 1:
```
>>> [f"{l:02}" for l in range(1, 13)]
['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12']
```
although `1, 13` might not be necessary if we don't care if a lot is named `"00"`, but the comment suggests otherwise

Reply:
Maybe also make it a variable, and use that in the assert too

Reply:
Also maybe don't use `l` as the loop variable so my eyes can read it properly :-)

Reply:
> We could just do this? Similarly for `lot_quantities` and in the asserts.
> 
> ```python
> ['1.02] * 12
> ```

I had a faint memory of repeating list element so...

> Or are we missing the `{}` in the f-strings?

Thanks on both counts.

Reply:
@PeterAlabaster I'm happy that this has been addressed - please resolve it if you are too.